### PR TITLE
feat(starfish): Add a percentile_percent_change

### PR DESCRIPTION
**Pull request by wmak**
- This adds a percentile_percent_change function which splits the current time period in half, calculates the percentiles for each half and deterimines the percent change
- Went with this name instead of percentile_delta since its more accurate (a delta is probably the difference between the halves)
- Needed to make alias optional since we don't need it for these sub percentiles

### CODE REVIEW

File: src/sentry/search/events/datasets/spans_metrics.py [hunk 1]
```diff
@@ -384,6 +398,54 @@ def _resolve_http_error_count(
...
+                    ],
+                ),
+            ],
+        )
+        return Function(
+            "multiply",
+            [
+                Function(
+                    "divide",
+                    [second_half, first_half],
```
Comment (shruthilayaj) [inline src/sentry/search/events/datasets/spans_metrics.py:442]: wait, don't we want to do `(second - first)/first` ?
this is just doing `second / first` right?

File: src/sentry/search/events/datasets/spans_metrics.py [hunk 2]
```diff
@@ -278,6 +278,20 @@ def function_converter(self) -> Mapping[str, fields.MetricsFunction]:
...
+                        fields.MetricArg(
+                            "column",
+                            allowed_columns=["span.duration"],
+                            allow_custom_measurements=False,
+                        ),
+                        fields.NumberRange("percentile", 0, 1),
+                    ],
+                    calculated_args=[resolve_metric_id],
+                    snql_distribution=self._resolve_percentile_percent_change,
+                    default_result_type="duration",
```
Comment (shruthilayaj) [inline src/sentry/search/events/datasets/spans_metrics.py:293]:
```suggestion
                    default_result_type="percentage",
```

File: src/sentry/search/events/datasets/spans_metrics.py [hunk 3]
```diff
@@ -384,6 +398,54 @@ def _resolve_http_error_count(
...
+            ],
+        )
+        return Function(
+            "multiply",
+            [
+                Function(
+                    "divide",
+                    [second_half, first_half],
+                ),
+                100,
```
Comment (shruthilayaj) [inline src/sentry/search/events/datasets/spans_metrics.py:444]: we don't normally do the multiply 100 on any of our rates right? we just return the ratio and convert to percentage in the fe

File: src/sentry/search/events/datasets/spans_metrics.py (hunk 1 above)
Comment (wmak) [reply src/sentry/search/events/datasets/spans_metrics.py:442]: Yep you're right I'll fix that :+1:

**Review by shruthilayaj — APPROVED** [review]
🚢

**Pull request merged**